docs/models_and_schedulers.rst

```diff
@@ -252,3 +252,146 @@ the following happens:
+ensemble where the stateful model in the ensemble must share the same flag.
+
+The :ref:`ensemble scheduler <section-ensemble-scheduler>` must be used for
+ensemble models, regardless of the scheduling choice of the model involved.
+
+As a running example, consider an ensemble model for image classification and
+segmentation that has the following model configuration::
+
+  name: "ensemble_model"
+  platform: "ensemble"
+  max_batch_size: 1
+  input [
+    {
+      name: "IMAGE"
+      data_type: TYPE_STRING
+      dims: [ 1 ]
+    }
+  ]
+  output [
+    {
+      name: "CLASSIFICATION"
+      data_type: TYPE_FP32
+      dims: [ 1000 ]
+      label_filename: "classification_labels.txt"
+    },
+    {
+      name: "SEGMENTATION"
+      data_type: TYPE_FP32
+      dims: [ 3, 224, 224 ]
+    }
+  ]
+  ensemble_scheduling {
+    step [
+      {
+        model_name: "image_preprocess_model"
+        model_version: -1
+        input_map {
+          key: "RAW_IMAGE"
+          value: "IMAGE"
+        }
+        output_map {
+          key: "PREPROCESSED_OUTPUT"
+          value: "preprocessed_image"
+        }
+      },
+      {
+        model_name: "classification_model"
+        model_version: -1
+        input_map {
+          key: "FORMATTED_IMAGE"
+          value: "preprocessed_image"
+        }
+        output_map {
+          key: "CLASSIFICATION_OUTPUT"
+          value: "CLASSIFICATION"
+        }
+      },
+      {
+        model_name: "segmentation_model"
+        model_version: -1
+        input_map {
+          key: "FORMATTED_IMAGE"
+          value: "preprocessed_image"
+        }
+        output_map {
+          key: "SEGMENTATION_OUTPUT"
+          value: "SEGMENTATION"
+        }
+      }
+    ]
+  }
+
+The ensemble_scheduling section indicates that the ensemble scheduler will be
+used and the ensemble model is consist of three different models. Each element
+in step section specifies the model to be used and how the inputs and outputs of
+the model are mapped to tensor names recognized by the scheduler. For
+example, the first element in step specifies that the latest version of
+image_preprocess_model should be used, the content of its input "RAW_IMAGE"
+is provided by "IMAGE" tensor, and the content of its output "PREPROCESSED_OUTPUT"
+will be mapped to "preprocessed_image" tensor for later use. The tensor names 
+recognized by the scheduler are the ensemble inputs, the ensemble outputs and
+all values in the input_map and the output_map.
+
+Assuming that only the ensemble model, the preprocess model, the classification
+model and the segmentation model are being served, the client applications will
+see them as four different models which can process requests independently.
+
+However, the ensemble scheduler will view the ensemble model as the following.
+
+.. image:: images/ensemble_example0.png
+
+When an inference request for the ensemble model is received, the ensemble
+scheduler will:
```

Comment:
First word for each list item is plural, so "will" should be omitted.

1. the ensemble scheduler will Recognizes ...
2. the ensemble scheduler will Checks models

vs:
1. the ensemble scheduler Recognizes ...
2. the ensemble scheduler Checks models